### PR TITLE
Add support for ServiceUnavailableRetryStrategy

### LINKED ISSUE #2093: Add ServiceUnavailableRetryStrategy to the client
HTTP Client allows you to setup a ServiceUnavailableRetryStrategy to deal with error responses from Origin servers. I'd like to add this to the Dropwizard client

Basically add this to HttpClientBuilder.java
```
    public HttpClientBuilder using(ServiceUnavailableRetryStrategy serviceUnavailableRetryStrategy) 
    {
        this.serviceUnavailableRetryStrategy = serviceUnavailableRetryStrategy;
        return this;
    }

    // ... then in createClient
 
    protected ConfiguredCloseableHttpClient createClient(
   
        ... snip ...

        if (serviceUnavailableRetryStrategy != null) {
            builder.setServiceUnavailableRetryStrategy(serviceUnavailableRetryStrategy);
        }


```

And the same sort of thing in JerseyClientBuilder.

### PR DESCRIPTION
**Pull request by patrox**
Resolves #2093.

### CODE REVIEW

File: dropwizard-client/src/test/java/io/dropwizard/client/HttpClientBuilderTest.java [hunk 1]
```diff
@@ -679,6 +680,19 @@ public void usesHttpProcessor() throws Exception {
...
 
     @Test
+    public void usesServiceUnavailableRetryStrategy() throws Exception {
+        ServiceUnavailableRetryStrategy serviceUnavailableRetryStrategy = mock(ServiceUnavailableRetryStrategy.class);
+        final ConfiguredCloseableHttpClient client =
+            builder.using(serviceUnavailableRetryStrategy)
+                .createClient(apacheBuilder, connectionManager, "test");
+        assertThat(client).isNotNull();
+        assertThat(FieldUtils.getField(httpClientBuilderClass,
+            "serviceUnavailStrategy", true)
```
Comment (patrox) [inline dropwizard-client/src/test/java/io/dropwizard/client/HttpClientBuilderTest.java]: This part was quite tricky, as internally it's named quite inconsistently :cry:
But I guess this is a punishment for forcing access to the field via reflection :wink:

**Comment by patrox** [comment]
@nickbabcock Yeah, sorry for that, after reinstalling IDEA my old settings disappeared :(
Should be fixed.

**Comment by arteam** [comment]
@patrox Sorry for nitpicking, but your last commit along with expanding added a lot of other formatting changes. Could you please revert it and just expand the imports?

**Comment by patrox** [comment]
@arteam Sure, sorry I missed that - otherwise I would fix it sooner ! Thanks for the nitpick :)

File: dropwizard-client/src/test/java/io/dropwizard/client/HttpClientBuilderTest.java (hunk 1 above)
Comment (arteam) [reply dropwizard-client/src/test/java/io/dropwizard/client/HttpClientBuilderTest.java]: I am also not a fan of unit tests with Reflection, especially considering most of them will fail on the new builds of Java 9, but in this case it should be good to go. Let's hope Apache Http Client will provide public accessors in future versions.

**Comment by arteam** [comment]
LGTM! Thanks @patrox and @nickbabcock!

**Pull request merged**